web/js/deerInitializer.js

```diff
@@ -390,13 +398,19 @@ function URIisValid(uriString) {
 }
 
 function setNamesTemplate(items,
-    trueTemplate = `<li><deer-view deer-id="$itemURI" deer-template="label"></deer-view></li>`,
+    trueTemplate = `<li><deer-view deer-id="$itemURI" deer-template="label">loading...</deer-view></li>`,
     falseTemplate = `<li> $itemURI </li>`) {
     let nameText = ``
     items.forEach((val) => {
-        const itemURI = UTILS.getValue(val)
-        const name = URIisValid(itemURI) ? trueTemplate.replaceAll("$itemURI",itemURI) : falseTemplate.replaceAll("$itemURI",itemURI)
-        nameText += name
+        //This should always be either the string value for the label or an item URI.  
+        //In most cases, we expect it will be a URI.
+        //If the item is blank or undefined, ignore it.  It is not real and should not be considered.  The citation source should be checked, it's value is off.
+        if(val && UTILS.getValue(val)){
```

Comment:
This is a lot...

* `UTILS.getValue(val)` will return `undefined` if `val` is nothing so this && check is extra.
* `getValue()` and `getLabel()` and `URIisValid()` are both called twice, but you can reduce it without breaking readability I think.

```js
const itemURI = UTILS.getValue(val)
const itemLabel = UTILS.getLabel(val) ?? itemURI ?? "Entity Not Labeled"
nameText += URIisValid(itemURI) ? trueTemplate.replaceAll("$itemURI",itemURI) : falseTemplate.replaceAll("$itemURI", itemLabel)
```

I don't think this breaks your intent and removes all duplicates. Maybe an undefined `val` sent in would make an extra unneeded template? Shouldn't need it, but an escape `if(!val) return` at the top could `continue` the loop without effect. Similarly, a `items.filter(val).forEach(` could prune the unhelpful records.